hazelcast/src/test/java/com/hazelcast/client/config/XmlClientConfigBuilderTest.java

```diff
@@ -687,74 +689,174 @@ public void testCompactSerialization() {
                 + "    </serialization>\n"
                 + HAZELCAST_CLIENT_END_TAG;
 
-        ClientConfig config = buildConfig(xml);
-        assertTrue(config.getSerializationConfig().getCompactSerializationConfig().isEnabled());
+        SerializationConfig config = buildConfig(xml).getSerializationConfig();
+        assertTrue(config.getCompactSerializationConfig().isEnabled());
     }
 
     @Override
-    public void testCompactSerialization_explicitSerializationRegistration() {
+    public void testCompactSerialization_serializerRegistration() {
         String xml = HAZELCAST_CLIENT_START_TAG
                 + "    <serialization>\n"
                 + "        <compact-serialization enabled=\"true\">\n"
-                + "            <registered-classes>\n"
-                + "                <class type-name=\"obj\" serializer=\"example.serialization.EmployeeDTOSerializer\">"
-                + "                    example.serialization.EmployeeDTO\n"
-                + "                </class>\n"
-                + "            </registered-classes>\n"
+                + "            <serializers>\n"
+                + "                <serializer>example.serialization.EmployeeDTOSerializer</serializer>\n"
+                + "            </serializers>\n"
                 + "        </compact-serialization>\n"
                 + "    </serialization>\n"
                 + HAZELCAST_CLIENT_END_TAG;
 
-        ClientConfig config = buildConfig(xml);
-        CompactTestUtil.verifyExplicitSerializerIsUsed(config.getSerializationConfig());
+        SerializationConfig config = buildConfig(xml).getSerializationConfig();
+        CompactTestUtil.verifyExplicitSerializerIsUsed(config);
```

Comment:
Valid point, I have updated the method to make use of a serializable object and add an assertion about it being serialized as compact

Reply:
Can we also somehow assert that reflective serializer is not used ? I think that's what we mean by "explicit" serializer is used.

Reply:
Since I made it Serializable, if the explicit serializer wouldn't be used, it wouldn't be Compact, as Java serializable has more priority over reflective serializer